hathor/transaction/script/opcode.py

```diff
@@ -0,0 +1,640 @@
+#  Copyright 2023 Hathor Labs
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import datetime
+import struct
+from enum import IntEnum
+from typing import Callable
+
+from cryptography.exceptions import InvalidSignature
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.asymmetric import ec
+
+from hathor.conf.get_settings import get_settings
+from hathor.crypto.util import (
+    get_address_b58_from_bytes,
+    get_hash160,
+    get_public_key_from_bytes_compressed,
+    is_pubkey_compressed,
+)
+from hathor.transaction.exceptions import (
+    EqualVerifyFailed,
+    InvalidScriptError,
+    InvalidStackData,
+    MissingStackItems,
+    OracleChecksigFailed,
+    ScriptError,
+    TimeLocked,
+    VerifyFailed,
+)
+from hathor.transaction.script.construct import (
```

Comment:
I feel all these methods should be in the `hathor.transaction.script.execution` module. Even though they are also used by parsers and constructors, they were developed to be used during script execution.

Reply:
Done!

Reply:
Thanks!